lib/pengine/group.c

```diff
@@ -249,7 +249,12 @@ pe__group_html(pcmk__output_t *out, va_list args)
         GListPtr rscs = pe__filter_rsc_list(rsc->children, only_rsc);
 
         if (rscs != NULL) {
-            out->begin_list(out, NULL, NULL, "Resource Group: %s", rsc->id);
+            if (pe__resource_is_disabled(rsc)) {
+                out->begin_list(out, NULL, NULL, "Resource Group: %s (disabled)", rsc->id);
+            } else {
+                out->begin_list(out, NULL, NULL, "Resource Group: %s", rsc->id);
+            }
```

Comment:
This is tempting but I think we need to check each individual group member -- target-role could be set individually in each one's meta-data rather than in the group meta-data.

I'm thinking of a pe__all_children_disabled(rsc) that would first check rsc's meta-data, then if necessary its children's. (Checking the members alone would be sufficient since they inherit the group meta-data, but I suspect short-circuiting the check is worthwhile for performance in large groups.)

Reply:
Hah.  I'd originally modified pe__resource_is_disabled to check the children resources, but backed all that out after I re-read the bug report and saw that it was looking for "if the group is disabled" and not "if all the children are disabled".  I should have stuck with my instincts.

Reply:
I would expect us to display "(disabled)" next to the group name if and only if either:

- `target-role=Stopped` in the group metadata, or
- `target-role=Stopped` in `rsc_defaults` and the group metadata does not override the default.

It looks like the current plan is to display "(disabled)" next to the group name in the above case **or** in case every child resource is disabled.

IMO it's inappropriate to display "(disabled)" next to the group name unless the whole group is disabled either explicitly in its own metadata, or implicitly through inheritance. I don't think we should do it if all the individual child resources within the group are disabled but the group itself is not disabled. To me, that's misleading.

Thoughts?

Reply:
I feel like an admin would want to know whether a resource is disabled or not regardless of how it got disabled, so I prefer just sticking "(disabled)" wherever target-role is effectively Stopped. It's a strong enough indicator to point someone to check the configuration, and it's clear if someone greps for either the group name or a member name. It would be nice to say "(disabled via resource defaults)" or whatever, but that would be considerably more complicated with the current code, so I don't think it's worth the effort.

If there were a difference in behavior between a disabled group and an enabled group with all its members disabled, then definitely it should be displayed differently, but as far as I know there isn't. Though I suppose there is the case where a group has an explicit target-role of Stopped and all its members override that with an explicit target-role of Started, but I would think any admin crazy enough to do that has to be intimately familiar with the details already.

BTW there is also the stop-all-resources cluster-wide property, which doesn't need to be part of this PR but would probably be nice to show somehow.

Reply:
> I feel like an admin would want to know... I don't think it's worth the effort.

Ack. This was discussed in the BZ but it's good to get the rationale out here publicly.

> If there were a difference in behavior between a disabled group and an enabled group with all its members disabled, then definitely it should be displayed differently, but as far as I know there isn't.

Alrighty, works for me. I don't know of any specific way that the behavior would differ, other than what happens if you then decide to run `pcs resource enable` on the group vs. on the child resources to undo the disabled state of the group. The config tells all.

> Though I suppose there is the case where a group has an explicit target-role of Stopped and all its members override that with an explicit target-role of Started, but I would think any admin crazy enough to do that has to be intimately familiar with the details already.

Either that, or they're confused or forgetful ;) And this also applies to "some" of its members overriding with target-role=Started, provided they start from the beginning of the group. The config (if the user takes a few seconds to view it) tells all, of course.

Incidentally, situations like this may become more common after we file a pcs RFE to make `pcs resource enable` set target-role=Started instead of unsetting target-role. As we discussed in the BZ, it seems like a good idea so that `pcs resource enable` can override `target-role=Stopped` in `rsc_defaults`. However, we'll want to give extra thought to pcs does when a user runs `pcs resource enable` on a child resource when its parent group is disabled, or `pcs resource disable` on a group when some of its child resources have target-role=Started explicitly set.